src/package/packageConvert.ts

```diff
@@ -122,3 +139,102 @@ async function createRequestObject(
   return requestObject;
 }
+
+async function pollForStatusWithInterval(
+  id: string,
+  retries: number,
+  packageId: string,
+  branch: string,
+  withProject: SfProject,
+  connection: Connection,
+  interval: Duration
+): Promise<PackageVersionCreateRequestResult> {
+  let remainingRetries = retries;
+  const pollingClient = await PollingClient.create({
+    poll: async (): Promise<StatusResult> => {
+      const results: PackageVersionCreateRequestResult[] = await pvcr.byId(id, connection);
+
+      if (_isStatusEqualTo(results, [Package2VersionStatus.success, Package2VersionStatus.error])) {
+        // complete
+        if (_isStatusEqualTo(results, [Package2VersionStatus.success])) {
+          // update sfdx-project.json
+          let projectUpdated = false;
+          if (withProject && !process.env.SFDX_PROJECT_AUTOUPDATE_DISABLE_FOR_PACKAGE_VERSION_CREATE) {
+            projectUpdated = true;
+            const query = `SELECT MajorVersion, MinorVersion, PatchVersion, BuildNumber FROM Package2Version WHERE Id = '${results[0].Package2VersionId}'`;
+            const packageVersionVersionString: string = await connection.tooling
+              .query<PackagingSObjects.Package2Version>(query)
+              .then((pkgQueryResult) => {
+                const record = pkgQueryResult.records[0];
+                return `${record.MajorVersion}.${record.MinorVersion}.${record.PatchVersion}-${record.BuildNumber}`;
+              });
+            const newConfig = await generatePackageAliasEntry(
+              connection,
+              withProject,
+              results[0].SubscriberPackageVersionId,
+              packageVersionVersionString,
+              branch,
+              packageId
+            );
+            withProject.getSfProjectJson().set('packageAliases', newConfig);
+            await withProject.getSfProjectJson().write();
+          }
+          await Lifecycle.getInstance().emit('Package/convert-success', {
+            id,
+            packageVersionCreateRequestResult: results[0],
+            projectUpdated,
+          });
+          return { completed: true, payload: results[0] };
+        } else {
+          let status = 'Unknown Error';
+          if (results?.length > 0 && results[0].Error.length > 0) {
+            const errors = [];
+            // for multiple errors, display one per line prefixed with (x)
+            if (results[0].Error.length > 1) {
+              results[0].Error.forEach((error) => {
+                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                errors.push(`(${errors.length + 1}) ${error}`);
+              });
+              errors.unshift(messages.getMessage('versionCreateFailedWithMultipleErrors'));
+            }
+            status = errors.length !== 0 ? errors.join('\n') : results[0].Error.join('\n');
+          }
+          await Lifecycle.getInstance().emit('Package/convert-error', { id, status });
```

Comment:
@WillieRuemmele would create exported constants, or enum or, string literal type so consumer can use those instead of having to known the exact string.